types/jsonwebtoken/index.d.ts

```diff
@@ -143,18 +144,18 @@ export declare function verify(
 /**
  * Asynchronously verify given token using a secret or a public key to get a decoded token
  * @param {String} token - JWT string to verify
- * @param {String|Buffer} secretOrPublicKey - Either the secret for HMAC algorithms, or the PEM encoded public key for RSA and ECDSA.
+ * @param {String|Buffer|Function} secretOrPublicKey - A string or buffer containing either the secret for HMAC algorithms, or the PEM encoded public key for RSA and ECDSA. If jwt.verify is called asynchronous, secretOrPublicKey can be a function that should fetch the secret or public key
  * @param {VerifyOptions} [options] - Options for the verification
  * @param {Function} callback - Callback to get the decoded token on
  */
 export declare function verify(
     token: string,
-    secretOrPublicKey: string | Buffer,
+    secretOrPublicKey: string | Buffer | Function,
```

Comment:
Please use a better type than function. https://github.com/DefinitelyTyped/DefinitelyTyped#common-mistakes

Reply:
Thanks for your feedback, new to this and learning. I've defined interfaces for each argument to the internal callback function as per the RFC for JWT token headers and from inspecting the library internally. I've tried not to be too strict on the arguments just in case there's a possible scenario where its too restrictive. Since that callback is user land code and there's not really great documentation on the library side for its usage I'm confident people will be doing all sorts of strange things. Thanks again.